apps/vscode/vscode.py

```diff
@@ -170,53 +170,183 @@ def command_palette():
         actions.user.vscode("workbench.action.moveEditorToLeftGroup")
 
-    def split_window_right():
+    def split_move_right():
         actions.user.vscode("workbench.action.moveEditorToRightGroup")
 
-    def split_window_up():
+    def split_move_up():
         actions.user.vscode("workbench.action.moveEditorToAboveGroup")
 
-    def split_window_vertically():
-        actions.user.vscode("workbench.action.splitEditor")
+    # Resizing
+    def split_toggle_orientation():
+        actions.user.vscode("workbench.action.toggleEditorGroupLayout")
 
-    def split_window():
-        actions.user.vscode("workbench.action.splitEditor")
+    def split_toggle_maximize():
+        actions.user.vscode("workbench.action.maximizeEditor")
+
+    def split_layout_reset():
+        actions.user.vscode("workbench.action.evenEditorWidths")
+
+    def split_expand():
+        actions.user.vscode("workbench.action.increaseViewWidth")
+        actions.user.vscode("workbench.action.increaseViewHeight")
+
+    def split_shrink():
+        actions.user.vscode("workbench.action.decreaseViewWidth")
+        actions.user.vscode("workbench.action.decreaseViewHeight")
+
+    def split_expand_width():
+        actions.user.vscode("workbench.action.increaseViewWidth")
+
+    def split_shrink_width():
+        actions.user.vscode("workbench.action.decreaseViewWidth")
+
+    def split_expand_height():
+        actions.user.vscode("workbench.action.increaseViewHeight")
+
+    def split_shrink_height():
+        actions.user.vscode("workbench.action.decreaseViewHeight")
 
-    # splits.py support end
 
+@ctx.action_class("user")
+class TabActions:
+    # Destruction
+
+    def tab_close_others():
+        actions.user.vscode("workbench.action.closeOtherEditors")
+
+    def tab_close_all():
+        actions.user.vscode("workbench.action.closeAllEditors")
+
+    # Navigation
+    def tab_focus_index(index: int):
+        if index < 10:
+            if is_mac:
+                actions.user.vscode_with_plugin(
+                    f"workbench.action.openEditorAtIndex{index}"
+                )
+            else:
+                actions.key(f"alt-{index}")
+        else:
+            actions.user.vscode_with_plugin("workbench.action.openEditorAtIndex", index)
+
+    def tab_focus_negative_index(index: int):
+        actions.user.vscode_with_plugin("workbench.action.openEditorAtIndex", -index)
+
+    def tab_focus_first():
+        if is_mac:
+            actions.key("cmd-1")
+        else:
+            actions.key("alt-1")
+
+    def tab_focus_final():
+        if is_mac:
+            actions.user.vscode("workbench.action.lastEditorInGroup")
+        else:
+            actions.key("alt-0")
+
+    # Arrangement
+    def tab_pin():
+        actions.user.vscode("workbench.action.pinEditor")
+
+    def tab_unpin():
+        actions.user.vscode("workbench.action.unpinEditor")
+
+    def tab_move_right():
+        actions.user.vscode("workbench.action.moveEditorRight")
```

Comment:
workbench.action.moveEditorToRightGroup

Reply:
I don't agree with this. tab_move_right() is meant to move a tab to the right, and in the case of vscode (imo) this should be inside the "group", rather than to a new group. That said, the action is still wrong as it should actually be suffixed with InGroup, so I've fixed them.

Moving a tab right within the group won't move to an adjacent group if that tab is in the right most index. You have to explicitly use moveEditorToRightGroup. I've added another set of actions tab_move_to_split_<direction>(), which will be applicable to vscode, jetbrains, and probably others, but not neovim, etc. 

Since tab_move_up() and tab_move_down() are also actually "group-based" I think, I've switched those to be tab_move_to_split_up() and tab_move_to_split_down() respectively.

I suppose worth noting, if you already have a tab at the right most position and you say tab move right you _might_ expect it to move to the left most index of the group to the right, but this isn't something I think we can easily do from just using available command ids alone. Probably would need a custom extension or something, so outside the scope of this PR.